stripe/src/main/java/com/stripe/android/AbstractEphemeralKey.java

```diff
@@ -81,6 +85,66 @@ private EphemeralKey(
         mType = type;
     }
 
+    protected AbstractEphemeralKey(
+            @Nullable JSONObject jsonObject
+    ) throws JSONException {
+        mCreated = jsonObject.getLong(FIELD_CREATED);
+        mExpires = jsonObject.getLong(FIELD_EXPIRES);
+        mId = jsonObject.getString(FIELD_ID);
+        mLiveMode = jsonObject.getBoolean(FIELD_LIVEMODE);
+        mObject = jsonObject.getString(FIELD_OBJECT);
+        mSecret = jsonObject.getString(FIELD_SECRET);
+
+        // Get the values from the associated objects array first element
+        JSONArray associatedObjectArray = jsonObject.getJSONArray(FIELD_ASSOCIATED_OBJECTS);
+        JSONObject typeObject = associatedObjectArray.getJSONObject(0);
+        mType = typeObject.getString(FIELD_TYPE);
+        mObjectId = typeObject.getString(FIELD_ID);
+    }
```

Comment:
This new constructor allows for better abstraction: a new class extending `AbstractEphemeralKey` would be able to implement that constructor and work with `EphemeralKeyManager`